crates/fluvio-spu/src/core/global_context.rs

```diff
@@ -478,5 +484,44 @@ mod file_replica {
                 error!("leader switch failed: {}", err);
             }
         }
+
+        /// Delete consumers offset for given replica if it is leader consumer
+        async fn delete_consumers_offset_if_leader_consumer(
+            &self,
```

Comment:
is this necessary? this only works if this SPU is leader is same as consumer but that may not case for multiple SPU

Reply:
but need to test case where consumer offset topic is not on same SPU